### PR TITLE
Adds missing tests

### PR DESCRIPTION
**Pull request by alexanderwallin**
This PR completes #27 by adding tests for:

* Compressor
* EnvelopeFilter
* LFO
* MoogFilter
* Phaser
* PingPongDelay
* Tremolo
* WahWah

It includes a couple of fixes that I think were plain bugs, or at least inconsistencies. Let me know if you find anything suspicious or if we should split this up into smaller PRs!

### CODE REVIEW

File: tuna.js [hunk 1]
```diff
@@ -2072,15 +2077,6 @@
                 this._target = value;
             }
         },
-        activate: {
```
Comment (Theodeus) [inline tuna.js]: **TL;DR**
I think we need to keep this activate method for the LFO. You can see that stuff breaks if we remove it by using a Chorus node which uses the LFO.

The reson why the activate method wasn't inherited for the LFO, as far as I can remember is that the LFO by default isn't connected to the output (rather assigned to control a audio param) and thus the script processor that it's made of doesn't do anything. The activate method removed here just jacks it into the context destination and the script processor starts processing.

In the LFO the activateNode and the output isn't connected to each other, and the reason for that is to make sure that the LFO script processor doesn't run if it doesn't have to, saving CPU cycles.

**Comment by Theodeus** [comment]
Wow. This is really an early christmas for me! I've been meaning to get around to fixing those tests for far too long now. Thanks a million times! 

There's just one detail with the LFO we need to fix, and then I'll merge this (and probably update to version 0.5.0).

Great work!

File: tuna.js (hunk 1 above)
Comment (alexanderwallin) [inline tuna.js]: Ah, yes that makes total sense. The only thing missing in the `activate()` override is the call to `activateCallback`, so I'll just put that in, ye?

Comment (Theodeus) [inline tuna.js]: Sounds good! :)

**Comment by alexanderwallin** [comment]
There we are. 🙂

**Comment by alexanderwallin** [comment]
I would argue this is a patch version update, though! Leading-zero versioning makes the minor part act as a major, and this PR does not introduce any breaking changes (or any changes at all, really) to consumers of the library.

**Comment by Theodeus** [comment]
True - was thinking of doing a minor version update with the other PR (once the Gain and Panner nodes are in). I'm looking at http://semver.org/, though I can't remember if this is the standard npm uses?

**Comment by alexanderwallin** [comment]
npm is using semver, yes!

https://docs.npmjs.com/getting-started/semantic-versioning

With leading zero, you kind of have to squash everything non-breaking into patches. Both non-breaking features and bug fixes are patches, basically.

**Comment by Theodeus** [comment]
Aha, gotcha, I think. So once we were at 1.0.0, adding the Gain and Panner would be a minor version, but while at 0.4.x you only do patch versions. Did I get it right?

**Comment by alexanderwallin** [comment]
Exactly!

**Comment by Theodeus** [comment]
Ok, cool. I'm thinking we might as well go 1.0.0 now then, and then follow semver. I've only been waiting for test coverage to jump there anyway!

**Pull request merged**